### PR TITLE
search list: add extend link when no result has been found

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Adds a link to extend the search to all documents (public and admin).
* Adds a control on document brief view if the document exists.
* Changes boolean to new observable on adminMode configuration.
* Improves dynamic routing system.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Dependencies

* rero/ng-core#278

## How to test?

- Search for a term that does not exist in the results (Ex: Cheval)
- See if the extend link is present
- Click on the link and see if the url has changed and if you have results.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

**Comment by jma** [comment]
The commit message should be updated. This concerns a search list when no result has been found.

**Review by jma — APPROVED** [review]

File: projects/public-search/src/app/document-record-search/document-record-search.component.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<ng-core-record-search [adminMode]="adminMode" [currentType]="currentType" [types]="types" [detailUrl]="detailUrl"
+  [showSearchInput]="showSearchInput" [q]="q" [page]="page" [size]="size" [sort]="sort"
+  (parametersChanged)="updateUrl($event)" (recordsSearched)="recordsSearched($event)"
+>
+  <div class="my-4" *ngIf="currentType === 'documents' && showLink">
+    <a href="#" (click)="linkToGlobalDocuments($event)" translate>
+      Expand to all RERO ILS
```
Comment (zannkukai) [inline projects/public-search/src/app/document-record-search/document-record-search.component.html:24]: Use the same message than previous : "Expand to all organisations" ? To be more generic

**Review by zannkukai — APPROVED** [review]

File: projects/admin/src/app/record/custom-editor/holding-editor/holding-editor.component.spec.ts [hunk 2]
```diff
@@ -7,34 +7,37 @@ import { TranslateService, TranslateModule, TranslateLoader, TranslateFakeLoader
...
+      params: { type: 'holdings', pid: 1 },
       data: {
         types: [
           {
             key: 'holdings',
           }
         ],
         showSearchInput: true,
-        adminMode: true
+        adminMode: of({ message: '', can: true })
```
[the remark this answers is not in the document]
Comment (Garfield-fr) [reply projects/admin/src/app/record/custom-editor/holding-editor/holding-editor.component.spec.ts]: done

File: projects/admin/src/app/record/document-record-search/document-record-search.component.spec.ts.disabled [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+import { ActivatedRoute, convertToParamMap, Router, UrlSegment } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+import { TranslateFakeLoader, TranslateLoader, TranslateModule } from '@ngx-translate/core';
+import { ActionStatus, DialogComponent, DialogService, RecordModule, RecordService } from '@rero/ng-core';
+import { Observable, of } from 'rxjs';
+import { ContributionFormatPipe } from '../../pipe/contribution-format.pipe';
+import { DocumentRecordSearchComponent } from './document-record-search.component';
+
+
+// FIX THIS TEST
```
[the remark this answers is not in the document]
Comment (Garfield-fr) [reply projects/admin/src/app/record/document-record-search/document-record-search.component.spec.ts.disabled:31]: done

File: projects/admin/src/app/record/brief-view/documents-brief-view/documents-brief-view.component.html [hunk 4]
```diff
@@ -15,36 +15,38 @@
...
-        {{ contribution | contributionFormat }}
-      </span>
-      <a *ngIf="contribution.agent.pid" [routerLink]="['/records', 'persons', 'detail', contribution.agent.pid]">
-        {{ contribution | contributionFormat }}
-      </a>
-      {{ last ? '' : '; ' }}
-      {{ last && record.metadata.contribution.length > 3 ? '; …' : '' }}
-    </li>
-  </ul>
+<ng-container *ngIf="record">
```
[the remark this answers is not in the document]
Comment (Garfield-fr) [reply projects/admin/src/app/record/brief-view/documents-brief-view/documents-brief-view.component.html:18]: done

File: projects/admin/src/app/record/custom-editor/holding-editor/holding-editor.component.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Garfield-fr) [reply projects/admin/src/app/record/custom-editor/holding-editor/holding-editor.component.spec.ts]: done

File: projects/admin/src/app/record/document-record-search/document-record-search.component.spec.ts.disabled (hunk 3 above)
[the remark this answers is not in the document]
Comment (Garfield-fr) [reply projects/admin/src/app/record/document-record-search/document-record-search.component.spec.ts.disabled:31]: done

File: projects/public-search/src/app/document-record-search/document-record-search.component.html (hunk 1 above)
Comment (Garfield-fr) [reply projects/public-search/src/app/document-record-search/document-record-search.component.html:24]: The message comes from Nicolas.
done

**Pull request merged**